rewrite some input rumble functions

punpckhdq opened the pull request:
more than 1 controller is untested but it will try its best to set the rumble state of extra controllers

punpckhdq committed:
rewrite some input rumble functions

punpckhdq committed:
appeasement

punpckhdq committed:
appeasement 2

Berthalamew commented:
Tested with kb/m and 2 controllers.
Vibration works correctly for controllers

Berthalamew commented on xlive/Blam/Engine/game/player_vibration.cpp:
```diff
@@ -33,133 +33,127 @@ void __cdecl rumble_player_continuous(int32 user_index, real32 rumble_intensity_
-            }
-        }
-
-        for (uint32 i = 0; i < k_number_of_users; i++)
```
why was this removed?

punpckhdq replied:
restored in latest commit

Berthalamew commented on xlive/Blam/Engine/input/input_xinput.cpp:
```diff
@@ -73,33 +72,44 @@ void input_xinput_clear_rumble_state(void)
-	if (controller_button_state_get(_controller_index_0)->plugged_in)
+	for (uint32 device_index = 0; device_index < k_number_of_controllers; ++device_index)
 	{
-		g_xinput_vibration.wLeftMotorSpeed = (suppress_rumble ? 0 : g_vibration_state[_controller_index_0].wLeftMotorSpeed);
```
rumble needs to be set to 0 if it's suppressed incase rumbling is already in progress.

punpckhdq replied:
handled in function

punpckhdq committed:
appeasement 3 and a bugfix
invert !g_input_feedback_suppress in input_xinput_update_rumble_state
eagerly set controller rumble states in vibration_update

Berthalamew commented:
Tested with kb/m and 3 controllers.
Vibration works correctly.
